runtime/near-vm-logic/src/logic.rs

```diff
@@ -943,14 +943,200 @@ impl<'a> VMLogic<'a> {
     pub fn keccak512(&mut self, value_len: u64, value_ptr: u64, register_id: u64) -> Result<()> {
         self.gas_counter.pay_base(keccak512_base)?;
         let value = self.get_vec_from_memory_or_register(value_ptr, value_len)?;
-        self.gas_counter.pay_per_byte(keccak512_byte, value.len() as u64)?;
+        self.gas_counter.pay_per(keccak512_byte, value.len() as u64)?;
 
         use sha3::Digest;
 
         let value_hash = sha3::Keccak512::digest(&value);
         self.internal_write_register(register_id, value_hash.as_slice().to_vec())
     }
 
+    /// Hashes the given value using RIPEMD-160 and returns it into `register_id`.
+    ///
+    /// # Errors
+    ///
+    /// If `value_len + value_ptr` points outside the memory or the registers use more memory than
+    /// the limit with `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    ///  Where `message_blocks` is `max(message_len / 8, 1)`.
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + ripemd160_base + ripemd160_block * message_blocks`
+    #[cfg(feature = "protocol_feature_evm")]
+    pub fn ripemd160(&mut self, value_len: u64, value_ptr: u64, register_id: u64) -> Result<()> {
+        self.gas_counter.pay_base(ripemd160_base)?;
+        let value = self.get_vec_from_memory_or_register(value_ptr, value_len)?;
+
+        let message_blocks = std::cmp::max(value_len / 8, 1);
```

Comment:
Ignoring leftover after `/ 8` feels odd here. Naively, I'd expect that we want [ceiling rather than floor](https://en.wikipedia.org/wiki/Floor_and_ceiling_functions) here. Might make sense to leave a comment explainign why we take floor here?